pkg/resources/apiserver/cabundle.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"fmt"
+
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type caBundleProvider interface {
+	CABundle() (string, error)
+}
+
+func CABundleCreator(data caBundleProvider) reconciling.NamedConfigMapCreatorGetter {
```

Comment:
Another duplicate of this function with slight vadiation for `caBundleProvider`.

Reply:
I see what you mean, but I'm not 100% sure that merging the various implementations into one just to save a few lines of code is the right way to go here. All of these are different, some work of a PEM-encoded CA bundle, some read files.

For example, the user-cltrmgr does not use `resources.TemplateData`, so creating a single CABundleCreatorGetter based on this would not work over there. We would need to have multiple, and we would also need to make the name configurable, because for the backup jobs the name is not simply `ca-bundle`. Currently we have:

* `func CABundleConfigMapCreator(caBundle *corev1.ConfigMap) reconciling.NamedConfigMapCreatorGetter {` because this copies another ConfigMap
* `func CABundleCreator(data caBundleProvider) reconciling.NamedConfigMapCreatorGetter` because this uses TemplateData
* `func caBundleConfigMapCreator(name string, caBundle string) reconciling.NamedConfigMapCreatorGetter` for the etcd backup ConfigMaps

What we end up with is IMHO needlessly genericized code that saves maybe 6 lines of code (plus some boilerplate, so make it 20).

Additionally, this would make later improvements harder, as we tied together things that were purposefully left somewhat separated: the CA bundle for userclusters is by design a different, but similar thing from the CA bundle of the seed.

TL;DR: I don't think it's worth to merge these.